setup.py

```diff
@@ -96,6 +96,14 @@ def get_all_extras() -> Dict:
 
 all_extras = get_all_extras()
 
+base_deps = [
+   *all_extras.get("crypto", []),
+    "PyYAML",
```

Comment:
I don't think we should include jsonschme, protobuf and PyYAML in the base version. Neither of them are used for programmatic usage.

Reply:
although you can decide whether the AEA can be run programmatically or not, the modules that use the package `yaml` are imported anyway. 

In order to remove those dependencies, the modules must be opportunely refactored. 

E.g. currently a chain of imports that also imports `yaml` is:
```
aea.aea -> needs Resources
aea.registries.base -> needs ConfigLoader
aea.configurations.loader -> needs yaml
```